language/move-prover/bytecode/src/read_write_set_analysis.rs

```diff
@@ -144,15 +146,17 @@ impl ReadWriteSetState {
         let callee_summary = callee_summary_.clone();
         // (1) bind all footprint values and types in callee locals to their caller values
         let mut new_callee_locals = callee_summary.locals.substitute_footprint(
-            actuals,
             type_actuals,
+            &temp_indices,
+            callee_fun_env,
```

Comment:
I think this should be `caller_fun_env`--we are replacing footprint values in the callee summary with caller state.

Reply:
I am not sure I understand why it should be the caller env rather than the callee's one.
The only place where this `FunctionEnv` is actually used is in `AccessPath::substitute_footprint` as an argument to `Root::from_index`, so to check whether the given index is a formal or a local.
I would think that we need the callee's env to know whether the given index is a formal or not. Or am I getting mixed up here?

Reply:
Here is my reasoning. If we have the following code:

```
fun caller(tmp1) {
  tmp2 = @0x2
  foo(tmp1, tmp2)
}
```

`temp_indices` should be `[tmp1, tmp2]`. When we get to `substitute_footprint`, we will need to resolve the values of `tmp1` and `tmp2` from the caller state. We do this by constructing an access path from `Root::from_index(tmp1, fun_env)`, then reading the value of that access path in the caller state. We want `tmp1` to be read as a `Formal` and `tmp2` to be read as a `Local`, and we need the caller's function environment to determine this.

Reply:
I see, that makes sense, thanks!